rust/crd/src/authentication.rs

```diff
@@ -1,20 +1,29 @@
-use serde::{Deserialize, Serialize};
-use snafu::{ResultExt, Snafu};
-use stackable_operator::commons::authentication::AuthenticationClassProvider;
+use std::future::Future;
+
+use snafu::{ensure, ResultExt, Snafu};
+use stackable_operator::commons::authentication::oidc::IdentityProviderHint;
+use stackable_operator::commons::authentication::{
+    ldap, oidc, static_, AuthenticationClassProvider, ClientAuthenticationDetails,
+};
 use stackable_operator::kube::ResourceExt;
 use stackable_operator::{
-    client::Client,
-    commons::authentication::AuthenticationClass,
+    client::Client, commons::authentication::AuthenticationClass,
     kube::runtime::reflector::ObjectRef,
-    schemars::{self, JsonSchema},
 };
+use tracing::info;
+
+use crate::NifiCluster;
+
+// The assumed OIDC provider if no hint is given in the AuthClass
+pub const DEFAULT_OIDC_PROVIDER: IdentityProviderHint = IdentityProviderHint::Keycloak;
+
+const SUPPORTED_OIDC_PROVIDERS: &[IdentityProviderHint] = &[IdentityProviderHint::Keycloak];
```

Comment:
I'm not sure, but should nifi-op care about the value of the providerHint at all?
https://github.com/stackabletech/operator-rs/blob/a796eed0fb4a70ced54027662fc7174b8474bce2/crates/stackable-operator/src/commons/authentication/oidc.rs#L78-L84

I though we only added this quirk for Superset, but I might be totally wrong here. It would be interesting what e.g. trino-op or druid-op are doing here